gabarit/template_vision/vision_project/package_name-scripts/1_preprocess_data.py

```diff
@@ -126,8 +127,12 @@ def main(directories: List[str], preprocessing: Union[str, None], sep: str = '{{
             # We start by creating a new directory for the preprocessed images
             new_directory = os.path.join(data_path, f"{directory}_{preprocess_str}")
             if os.path.exists(new_directory):
-                shutil.rmtree(new_directory)
-            os.makedirs(new_directory)
+                if overwrite:
+                    shutil.rmtree(new_directory)
+                    os.makedirs(new_directory)
+                else:
+                    logger.info(f"{new_directory} already exists. Pass.")
+                    continue
```

Comment:
Don't we need to create the directory if it does not exists ?
We might do it later, but in this case we don't need `os.makedirs(new_directory)` line 132 neither.